Sources/Calendar.php

```diff
@@ -293,6 +293,8 @@ function CalendarPost()
 		'%T' => '%l:%M',
 	));
 
+	$time_string = trim($time_string, ':');
```

Comment:
```suggestion
	$time_string = preg_replace('~:(?=\D|$)~', '', $time_string);
```

This will remove any colon that is followed by a non-digit or the end of the string.

| Before | After |
|--------|-------|
| '9:45:00 am' | '9:45:00 am' |
| '9:45: am' | '9:45 am' |
| '9:45:am' | '9:45am' |
| '9:: am' | '9 am' |
| '9::am' | '9am' |

Reply:
The time_string contains the time format, so no digits. But I think I managed to modify it to work.

Reply:
Derp. You are right. See comments below.